tools/c7n_azure/c7n_azure/query.py

```diff
@@ -42,8 +42,14 @@ def filter(self, resource_manager, **params):
         if extra_args:
             params.update(extra_args)
 
-        op = getattr(getattr(resource_manager.get_client(), enum_op), list_op)
-        data = [r.serialize(True) for r in op(**params)]
+        data = []
+        try:
+            op = getattr(getattr(resource_manager.get_client(), enum_op), list_op)
+            data = [r.serialize(True) for r in op(**params)]
+        except Exception as e:
+            log.error("Failed to query resource.\n"
+                      "Type: azure.{0}.\n"
+                      "Error: {1}".format(resource_manager.type, e))
```

Comment:
We do not fail gracefully when I `sys.exit` here. Like I said in the description, "If an exception is thrown, the data return empty. Consequently, in the logs it will show zero resources were fetched."

Reply:
Yeah, but it treats it as a successful execution (if you run c7n via the script e.g.)

Reply:
Ok I agree custodian should exit in error state here.  In Python 3 right thing to do is after log reraise with:
`raise Exception from e` (or whatever exception type you think is appropriate) - and this will give you the correct stack as if you had never caught it.

If we are supporting Python 2 for another month or so you have to do some nonsense like this:
`raise MyCustomException, MyCustomException(e), sys.exc_info()[2]`

Or both Six and Python Future has Raise_from for Py2 (I think we don't use six anymore?) 
http://python-future.org/compatible_idioms.html?highlight=raise_from#raising-exceptions

Reply:
Agreed. I will reraise the exception.